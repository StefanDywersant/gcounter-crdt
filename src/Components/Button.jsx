import React from 'react';
import PropTypes from 'prop-types';
import styles from './Button.scss';

/**
 * Renders that hacker-ish button.
 */
const Button = ({className, onClick, children}) => (
	<button
		onClick={onClick}
		className={[styles.button, className].join(' ')}
	>
		{children}
	</button>
);


Button.propTypes = {

	/**
	 * Optional class name for out element
	 */
	className: PropTypes.string,

	/**
	 * Buttons click handler
	 */
	onClick: PropTypes.func,

	/**
	 * Button's inner content
	 */
	children: PropTypes.node

};


export default Button;