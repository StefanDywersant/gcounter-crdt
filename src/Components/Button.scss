.button {
	border: 1px solid #0f0;
	background: #000;
	color: #0f0;
	font: 12px monospace;
	border-radius: 6px;
	outline: none;

	&:active {
		background: #0f0;
		color: #000;
	}
}