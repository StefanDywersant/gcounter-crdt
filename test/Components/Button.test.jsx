import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import Button from "../../src/Components/Button";

const renderer = new ReactShallowRenderer();

describe('Component rendering', () => {

	it('renders correctly with default props', () => {
		expect(renderer.render(
			<Button />
		)).toMatchSnapshot();
	});

	it('renders correctly with className set', () => {
		expect(renderer.render(
			<Button className="customClassName" />
		)).toMatchSnapshot();
	});

	it('renders correctly with children', () => {
		expect(renderer.render(
			<Button>Test button</Button>
		)).toMatchSnapshot();
	});

});
