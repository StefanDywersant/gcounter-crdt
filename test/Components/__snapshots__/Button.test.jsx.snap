// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component rendering renders correctly with children 1`] = `
<button
  className="button "
  onClick={undefined}
>
  Test button
</button>
`;

exports[`Component rendering renders correctly with className set 1`] = `
<button
  className="button customClassName"
  onClick={undefined}
/>
`;

exports[`Component rendering renders correctly with default props 1`] = `
<button
  className="button "
  onClick={undefined}
/>
`;
